app/ide-desktop/lib/dashboard/src/components/AriaComponents/Dialog/Dialog.tsx

```diff
@@ -150,6 +152,11 @@ export function Dialog(props: types.DialogProps) {
                   <div className="relative flex-auto overflow-y-auto p-3.5">
                     {typeof children === 'function' ? children(opts) : children}
                   </div>
+                  {closeButton === 'floating' && (
+                    <div className="m-floating-buttons absolute flex gap-1">
```

Comment:
style deduplication should not be a concern though

Reply:
It's always a concern

Reply:
Because:

1. `!important` hell
2. m-floating-buttons - how'd I understand how much it is in pixels?
3. tw-merge doesn't know `m-floating-buttons` 
4. We use the atomic approach, and style deduplication is one of the most important features.
5. We have tailwind-variants - if you want to add styles like 'button' use this lib instead.

Reply:
> `!important` hell

should not happen, an `m-` is still an `m-`.

> how'd I understand how much it is in pixels?

you're not supposed to

> tw-merge doesn't know `m-floating-buttons`

however `tw-merge` is trivial to configure so this is not an issue in practice

> style deduplication is one of the most important features

not too sure about this. because of variants, we probably end up with much more generated CSS.

> We have `tailwind-variants`

not sure how this is different to (or better than) using `@apply` directly


also remember: the goal of tailwind is not to golf your css. the goal of tailwind is not to deduplicate styles. the goal of tailwind is to encode a design system, and arbitrary pixel values are not a design system.

Reply:
> not too sure about this. because of variants, we probably end up with much more generated CSS.

It's not

> not sure how this is different to (or better than) using @apply directly

Apply is considered a bad practice by the authors of tw.

> you're not supposed to
Yes we supposed to. Because we. read. the. code.

Reply:
We're not going to use `m-floating-buttons`, `button` or any other classes. Period